Adds aggregate wmu column and command

noahcaldwell7 opened the pull request.

noahcaldwell7 commented:
Resolves #432 and #433, with the exception of filters by township (as I'm not sure we track that) or by year (need more detail from Kristi). Ready for review.

almasaeed2010 commented on app/Http/Controllers/ReportController.php:
```diff
@@ -0,0 +1,118 @@
+                    ->groupBy('aggregate_wmu')
+                , 'view')
+            ->groupBy('aggregate_wmu')
+            ->orderBy($request->order_by ?? 'aggregate_wmu', $request->order_dir ?? 'asc')
```
Looks like order_dir is not validated. You can either validate the input or use this strategy instead:
```suggestion
            ->orderBy($request->order_by ?? 'aggregate_wmu', $request->order_dir === 'asc' ? 'asc' : 'desc')
```
This helps fend off SQL injection and makes sure typos in JS won't completely break the page.

almasaeed2010 commented on app/Http/Controllers/ReportController.php:
```diff
@@ -0,0 +1,118 @@
+        ])
+            ->has('sites')
+            ->withCount(['sites'])
+            ->orderBy($request->order_by ?? 'name', $request->order_dir ?? 'asc')
```
validate order_dir.

almasaeed2010 commented on app/Http/Controllers/SiteController.php:
```diff
@@ -286,4 +289,74 @@ public function changeOwner(Site $site, Request $request)
+     */
+    public function layers(): string
+    {
+        return Storage::get("maps/results.geojson");
```
which layer is this? aggregate wmus or just wmus?

almasaeed2010 commented on resources/js/components/Reports/ReportByAggregateWmu.vue:
```diff
@@ -0,0 +1,134 @@
+
+        this.sites = data.data
+      } catch (e) {
+        if (!axios.isCancel()) {
```
As an improvement to error handling here we can check for status 422 and alert the user that there is an input error. This would help us debug reported issues faster too. I'd add something like this everywhere we could receive a validation error:

```js
if (!axios.isCancel()) {
  if(e.response && e.response.status === 422) {
    // TODO: import Errors from '../../forms/Errors'
    const errors = new Errors(e.response.data)
    this.$alert(errors.toArray().join(' ')
  } else {
    this.$alert('Unable to load sites. Please try refreshing the page or contact us.')
  }
  console.error(e)
}

almasaeed2010 reviewed (changes requested):
Looks great! Mostly minor updates 👍

noahcaldwell7 commented on app/Http/Controllers/ReportController.php:
```diff
@@ -0,0 +1,118 @@
+
+class ReportController extends Controller
+{
+    protected static array $aggregate_wmus = [
+        'Adirondacks', 'Catskills', 'Central Appalachian Plateau', 'Central Finger Lakes', 'Central NY',
+        'Closed', 'Del-Otsego', 'Del-Sullivan', 'E Appalachian Plateau', 'E Lake Plains', 'Mid Lake Plains',
+        'Mohawk Valley', 'NE Appalachian Hills', 'NE Hudson', 'NW Appalachian Hills', 'NW Hudson', 'SE Hudson',
+        'St. Lawrence Valley', 'Suffolk - Westchester', 'SW Hudson', 'W Appalachian Hills', 'W Appalachian Plateau',
+        'W Finger Lakes', 'W Lake Plains',
+    ];
```
I was originally using it before we queried aggregate_wmus. After we switched to using a query, I left the array because I thought a future feature might make use of it.

noahcaldwell7 commented:
Ready for further review. I made the requested changes, fixed the maps page, added a tab in the Reports page for breakdown by Town, and added pagination and search to the Town and State reports.

almasaeed2010 commented on app/Http/Controllers/ReportController.php:
```diff
@@ -0,0 +1,149 @@
+        ]);
+        $this->authorize('viewAny', Site::class);
+
+        $sites = \DB::table('sites')
```
```suggestion
        $sites =Site::query()
```
